rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt

```diff
@@ -1302,17 +1302,24 @@ data class DefineStmt(
         get() = "DEFINE"
 
     override fun dataDefinition(): List<InStatementDataDefinition> {
-        // FIXME: If indicator was not priorly SET (SETON or SETOFF), program should error
+        val containingCU = this.ancestor(CompilationUnit::class.java)
+            ?: return emptyList()
+
         val indicatorPattern = Regex("\\*IN\\d\\d")
-        val isIndicator = originalName.trim().uppercase().matches(indicatorPattern)
+        val normalizedOriginalName = originalName.trim().uppercase()
+        val isIndicator = normalizedOriginalName.matches(indicatorPattern)
         if (isIndicator) {
+            val indicatorKey = normalizedOriginalName.removePrefix("*IN").toIndicatorKey()
+            val setStatements = containingCU.main.stmts.explode(true).filterIsInstance<SetStmt>()
+            val definedIndicators = setStatements.map { it.indicators }.flatten().filterIsInstance<IndicatorExpr>()
+            val isIndicatorDefined = definedIndicators.any { it.index == indicatorKey }
+
+            if (!isIndicatorDefined) throw Error("Data reference $originalName not resolved")
```

Comment:
I used it to keep consistency with what is already present in `DefineStmt` logic. Do you think it would be wise to also update already existing occurrencies with the same logic?

Reply:
Sorry you are right.
Let's maintain the "status quo".